### PR TITLE
Sorting alphabetically subfolders in Content Search

### LINKED ISSUE #24482: Subfolders in Site Browser are not sorted alphabetically if the folder Name is changed #20104
### Parent Issue

_No response_

### Problem Statement

Subfolders in Site Browser are not sorted alphabetically if the folder Name is changed #20104

### Steps to Reproduce

To Reproduce
Create a folder at root level
Create a few sub folders where the Titles and Names agree
They appear to be sorted alphabetically sorted
Add another folder where the Title and Name are different
The folder is sorted by Title not Name
Browser folder tree
Site or Folder tree

reported and verified in https://dotcms.zendesk.com/agent/tickets/110684

### Acceptance Criteria

all folders even if renamed should be  sorted alphabetically 

### dotCMS Version

PLATFORM EDITION: 23.01

### Proposed Objective

Customer Support

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

reported and verified in https://dotcms.zendesk.com/agent/tickets/110684

https://dotcms.slack.com/archives/C028Z3R2D/p1679676989653299

### Assumptions & Initiation Needs

reported fixed in https://github.com/dotCMS/core/issues/20104
but this view was over looked
Site or Folder tree

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by AndreyDotcms**
### Summary
📁🔤🛠️

Sort subfolders by name in the browser portlet. This change fixes the issue by adding a line of code to `BrowserAjax.java` that sorts the subfolders list before returning it.

### Walkthrough
* Sort the subfolders list by folder name to fix issue

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java [hunk 1]
```diff
@@ -2270,6 +2270,8 @@ public List<Map<String, Object>> getFolderSubfolders(final String parentFolderId
 		final Role[] roles = DwrUtil.getUserRoles(user);
 		final Folder parentFolder = this.folderAPI.find(parentFolderId, user, false);
 		final List<Folder> subFolders = this.folderAPI.findSubFolders(parentFolder, user, false);
+		// Sort the list by folder name
+		subFolders.sort(Comparator.comparing(Folder::getName));
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java:2274]: Need to also make sure this work is in the BrowserResource or we will have a regression as soon as we rewrite the browser ui.

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java:2274]: @AndreyDotcms let's try to do a postman test... if you have any questions don't hesitate to reach me.

Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java]: Donde https://github.com/dotCMS/core/pull/26281/commits/352e76585600ab84437be63f5fe246077943cf9d

**Review by jcastro-dotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by wezell — APPROVED** [review]
Looks good, thanks guys

**Pull request merged**